### PR TITLE
Bundle Host FXR MSI

### PR DESCRIPTION
**Pull request by schellap**
Bring in hostfxr installer as well into CLI bundle.

@eerhardt @brthor could you pls. push into this branch for osx deb packages?

### CODE REVIEW

**Comment by schellap** [comment]
@brthor I've cherry picked your commit from brthor/cli@a8da901, thank you!

Also, PTAL.

**Comment by schellap** [comment]
I've fixed the description of hostfxr from "The hostFXR" to something better ;)

File: packaging/osx/clisdk/Distribution-Template [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <choice id="{SharedFxComponentId}.pkg" visible="true" title="{SharedFxBrandName} (x64)" description="The .NET Core Shared Framework">
         <pkg-ref id="{SharedFxComponentId}.pkg" />
     </choice>
-    <choice id="{HostFxrComponentId}.pkg" visible="true" title="{HostFxrBrandName} (x64)" description="The .NET Core HostFxr">
+    <choice id="{HostFxrComponentId}.pkg" visible="true" title="{HostFxrBrandName} (x64)" description="The .NET Core Host FX Resolver">
```
Comment (brthor) [inline packaging/osx/clisdk/Distribution-Template]: good idea.

File: build_projects/dotnet-cli-build/CliDependencyVersions.cs [hunk 2]
```diff
@@ -7,10 +7,12 @@ namespace Microsoft.DotNet.Cli.Build
 {
     public class CliDependencyVersions
     {
-        public static readonly string SharedFrameworkVersion = "1.0.0-rc3-004459-00";
-        public static readonly string SharedHostVersion = "1.0.1-rc3-004459-00";
+        public static readonly string SharedFrameworkVersion = "1.0.0-rc3-004465-00";
+        public static readonly string SharedHostVersion = "1.0.1-rc3-004465-00";
+        public static readonly string HostFxrVersion = "1.0.1-rc3-004465-00";
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: This needs to be updated by update-dependencies.  See https://github.com/dotnet/cli/blob/rel/1.0.0-preview2/build_projects/update-dependencies/UpdateFilesTargets.cs#L221

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs [hunk 3]
```diff
@@ -13,6 +13,7 @@ public class Monikers
         public const string CLISdkBrandName = "Microsoft .NET Core 1.0.0 - SDK Preview 2";
         public const string SharedFxBrandName = "Microsoft .NET Core 1.0.0 - Runtime";
         public const string SharedHostBrandName = "Microsoft .NET Core 1.0.0 - Host";
+        public const string HostFxrBrandName = "Microsoft .NET Core 1.0.0 - Host FX Resolver";
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: It just seems odd to me that these strings are in CLI.  Shouldn't they be coming from the packages built in core-setup?

Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: /cc @brthor

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs [hunk 4]
```diff
@@ -93,6 +71,11 @@ public static string GetSdkDebianPackageName(BuildTargetContext c)
             return $"{packagePrefix}-dev-{nugetVersion}";
         }
 
+        public static string GetDebianHostFxrPackageName(BuildTargetContext c)
```
Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Is this called?  I don't see it being invoked.

Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: this isn't called here. added for consistency but we could remove it

Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Let's remove it if it isn't called.

core-setup should have it's own "HostMonikers" class.  I really don't like putting non-shared things in this 'shared-utils' and then trying to keep them in sync.  Anyone coming in here will see this dead code and will remove it.

Comment (schellap) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: It should be called. I just updated it. The CI failed because of not calling it.

Comment (eerhardt) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Oh, I see it being called now with @schellap's latest commit.

File: build_projects/dotnet-cli-build/CliDependencyVersions.cs (hunk 2 above)
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: Is this going to be taken care of?

Comment (eerhardt) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: It is needed for the next time we update to a new build (which will be in like an hour)

Comment (schellap) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: I will wait for the dotnet-bot to pick this up and merge this with whatever new version it brings in.

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs (hunk 4 above)
Comment (schellap) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Btw, good call.

File: build_projects/dotnet-cli-build/CliDependencyVersions.cs (hunk 2 above)
Comment (eerhardt) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: dotnet-bot doesn't know how to "pick this up" without you updating that file I pointed to above.

Comment (eerhardt) [inline build_projects/dotnet-cli-build/CliDependencyVersions.cs]: We can't be adding new places with hard-coded versions without updating the 'update-dependencies' script.

File: build_projects/update-dependencies/UpdateFilesTargets.cs [hunk 5]
```diff
@@ -221,6 +221,7 @@ public static BuildTargetResult ReplaceDependencyVersions(BuildTargetContext c)
             ReplaceFileContents(@"build_projects\dotnet-cli-build\CliDependencyVersions.cs", fileContents =>
             {
                 fileContents = ReplaceDependencyVersion(c, fileContents, "SharedFrameworkVersion", "Microsoft.NETCore.App");
+                fileContents = ReplaceDependencyVersion(c, fileContents, "HostFxrVersion", "Microsoft.NETCore.DotNetHostResolver");
```
Comment (eerhardt) [inline build_projects/update-dependencies/UpdateFilesTargets.cs]: :+1:

**Comment by eerhardt** [comment]
:shipit: when green

**Comment by eerhardt** [comment]
I assume we don't need approval for this, since this is a MUST HAVE in order to consume the latest Shared FX.  But @schellap can you send an email to netcoreship anyway, just so they are aware?

@Petermarcu @piotrpMSFT

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs (hunk 3 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Yeah they should be. There's not a good way to transfer the info in the packages themselves that I know of though. 
Perhaps we could add them to the latest file in some way?

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs (hunk 4 above)
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: this is not correct any longer. Please take the latest of this method that I have merged in core-setup

Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: the version of hostfxr is included in the name.

**Comment by brthor** [comment]
We must update the hostfxr deb name before merging. That was part of the issue with SxS hostfxr packages in core-setup

**Comment by brthor** [comment]
:shipit:

**Comment by schellap** [comment]
@brthor updating to latest core-setup latest packages causes some unrelated test failures. I am going to sync to 4465 which gives me a green CI. The 4465 doesn't have your deb versioning change. I will file a bug to get the Deb packages updated to have a version. The Deb package version update requires getting latest stable versions and I don't want to hold up with check-in for that.

**Comment by brthor** [comment]
@schellap `4465` deb packages are not only unversioned but broken.

**Comment by schellap** [comment]
@brthor yes, I realized after posting that whatever unversioned is in Azure is not necessarily 4465. So I aborted my attempts to get 4465 dependencies.

**Comment by schellap** [comment]
@brthor, @eerhardt this PR is blocked on needing a `dotnet-compile-fsc` package targeting 1.0.0 `NETCore.App` here: https://dotnet.myget.org/feed/cli-deps/package/nuget/dotnet-compile-fsc

The failure is due to this in the `runtimeconfig.json`:
`"Microsoft.NETCore.App": "1.0.0-rc2-3002678"`

There is no roll-forward into RTM from pre-release.

```
20:14:05 The specified framework 'Microsoft.NETCore.App', version '1.0.0-rc2-3002678' was not found.
20:14:05   - Check application dependencies and target a framework version installed at:
20:14:05       /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0-preview2/debug_debian8.2_x64_prtest/artifacts/debian.8-x64/stage2/shared/Microsoft.NETCore.App
20:14:05   - The following versions are installed:
20:14:05       1.0.0
20:14:05   - Alternatively, install the framework version '1.0.0-rc2-3002678'.
```

**Comment by brthor** [comment]
where is rc2 coming from .. hmm

**Comment by brthor** [comment]
I found the issue @schellap the test assets are not using the currently built fsc

**Comment by schellap** [comment]
Closing... ported over by @brthor to #3555

**Pull request closed without merging**